cvat/apps/engine/model_utils.py

```diff
@@ -18,3 +22,86 @@ class Undefined:
 The reverse side of one-to-one relationship.
 May be undefined in the object, should be accessed via getattr().
 """
+
+
+_ModelT = TypeVar("_ModelT", bound=models.Model)
+
+
+def bulk_create(
+    db_model: type[_ModelT],
+    objs: Iterable[_ModelT],
+    batch_size: int | None = ...,
+    ignore_conflicts: bool = ...,
+    update_conflicts: bool | None = ...,
+    update_fields: Sequence[str] | None = ...,
+    unique_fields: Sequence[str] | None = ...,
+    *,
+    flt_param: dict[str, Any] | None = None,
+) -> list[_ModelT]:
+    "Like Django's Model.objects.bulk_create(), but applies the default batch size"
+
+    if batch_size is Ellipsis:
+        batch_size = settings.DEFAULT_DB_BULK_CREATE_BATCH_SIZE
+
+    kwargs = {}
+    for k, v in {
+        "ignore_conflicts": ignore_conflicts,
+        "update_conflicts": update_conflicts,
+        "update_fields": update_fields,
+        "unique_fields": unique_fields,
+    }.items():
+        if v is not Ellipsis:
+            kwargs[k] = v
+
+    if not objs:
+        return []
+
+    flt_param = flt_param or {}
+
+    if flt_param:
+        if "postgresql" in settings.DATABASES["default"]["ENGINE"]:
+            return db_model.objects.bulk_create(objs, batch_size=batch_size, **kwargs)
+        else:
+            # imitate RETURNING
```

Comment:
I wonder if this workaround can be removed now? The only relevant database that this would be used for is SQLite; according to the Django docs, SQLite 3.35 supports returning the primary key; and Ubuntu 22.04 already has SQLite 3.37.

Reply:
Yes, probably.